Flaky test learning_resources.etl.loaders_test. test_load_program

<!--- Provide a general summary of the issue in the Title above -->

### Expected Behavior
CI tests should consistently pass

### Current Behavior
This test had a recent test failure on CI:

> FAILED learning_resources/etl/loaders_test.py::test_load_program[False-True-False-True] - learning_resources.models.LearningResourceImage.MultipleObjectsReturned: get() returned more than one LearningResourceImage -- it returned 2!


### Steps to Reproduce
Run  `.manage.py pytest learning_resources/etl/loaders_test.py` multiple times, all tests should pass every time, particularly the `test_load_program` test.

--- a/learning_resources/etl/loaders.py
+++ b/learning_resources/etl/loaders.py
@@ -123,8 +123,10 @@ def load_instructors(
 def load_image(resource: LearningResource, image_data: dict) -> LearningResourceImage:
     """Load the image for a resource into the database"""
     if image_data:
-        image, _ = LearningResourceImage.objects.get_or_create(**image_data)
-
+        if LearningResourceImage.objects.filter(**image_data).exists():
+            image = LearningResourceImage.objects.filter(**image_data).first()
+        else:
+            image = LearningResourceImage.objects.create(**image_data)
         resource.image = image
     else:
         resource.image = None

--- a/learning_resources/etl/loaders.py
+++ b/learning_resources/etl/loaders.py
@@ -123,10 +123,11 @@ def load_instructors(
 def load_image(resource: LearningResource, image_data: dict) -> LearningResourceImage:
     """Load the image for a resource into the database"""
     if image_data:
-        if LearningResourceImage.objects.filter(**image_data).exists():
-            image = LearningResourceImage.objects.filter(**image_data).first()
-        else:
-            image = LearningResourceImage.objects.create(**image_data)
+        image, _ = LearningResourceImage.objects.get_or_create(
+            url=image_data.get("url"),
+            description=image_data.get("description"),
+            alt=image_data.get("alt"),
+        )
         resource.image = image
     else:
         resource.image = None

--- a/learning_resources/etl/loaders_test.py
+++ b/learning_resources/etl/loaders_test.py
@@ -23,6 +23,7 @@ from learning_resources.etl.loaders import (
     load_content_files,
     load_course,
     load_courses,
+    load_image,
     load_instructors,
     load_offered_by,
     load_podcast,
@@ -51,6 +52,7 @@ from learning_resources.models import (
     ContentFile,
     Course,
     LearningResource,
+    LearningResourceImage,
     LearningResourceOfferor,
     LearningResourceRun,
     PodcastEpisode,
@@ -679,6 +681,32 @@ def test_load_content_file():
         assert (
             getattr(loaded_file, key) == value
         ), f"Property {key} should equal {value}"
+
+
+def test_load_image():
+    """Test that image resources are uniquely created or retrieved based on parameters"""
+    LearningResourceImage.objects.filter(url="https://mit.edu").delete()
+    learning_resource = LearningResourceFactory.create()
+
+    # first image should be different from second due to 'description' parameter
+    image_a = load_image(learning_resource, image_data={"url": "https://mit.edu"})
+    image_b = load_image(
+        learning_resource, image_data={"url": "https://mit.edu", "description": ""}
+    )
+    assert image_a.id != image_b.id
+
+    # first image should be the same as third image since url and alt field matches
+    image_c = load_image(
+        learning_resource, image_data={"url": "https://mit.edu", "alt": None}
+    )
+    assert image_a.id == image_c.id
+
+    # fourth image should have a totally new id since all fields are unique
+    image_d = load_image(
+        learning_resource,
+        image_data={"url": "https://mit.edu", "alt": "test", "description": "new"},
+    )
+    assert image_d.id not in {image_a.id, image_b.id, image_c.id}
 
 
 def test_load_content_file_error(mocker):

--- a/learning_resources/etl/loaders_test.py
+++ b/learning_resources/etl/loaders_test.py
@@ -685,26 +685,27 @@ def test_load_content_file():
 
 def test_load_image():
     """Test that image resources are uniquely created or retrieved based on parameters"""
-    LearningResourceImage.objects.filter(url="https://mit.edu").delete()
+    resource_url = "https://mit.edu"
+    LearningResourceImage.objects.filter(url=resource_url).delete()
     learning_resource = LearningResourceFactory.create()
 
     # first image should be different from second due to 'description' parameter
-    image_a = load_image(learning_resource, image_data={"url": "https://mit.edu"})
+    image_a = load_image(learning_resource, image_data={"url": resource_url})
     image_b = load_image(
-        learning_resource, image_data={"url": "https://mit.edu", "description": ""}
+        learning_resource, image_data={"url": resource_url, "description": ""}
     )
     assert image_a.id != image_b.id
 
     # first image should be the same as third image since url and alt field matches
     image_c = load_image(
-        learning_resource, image_data={"url": "https://mit.edu", "alt": None}
+        learning_resource, image_data={"url": resource_url, "alt": None}
     )
     assert image_a.id == image_c.id
 
     # fourth image should have a totally new id since all fields are unique
     image_d = load_image(
         learning_resource,
-        image_data={"url": "https://mit.edu", "alt": "test", "description": "new"},
+        image_data={"url": resource_url, "alt": "test", "description": "new"},
     )
     assert image_d.id not in {image_a.id, image_b.id, image_c.id}
 

